test/e2e/performanceprofile/functests/utils/nodes/nodes.go

```diff
@@ -104,6 +105,24 @@ func GetBySelector(selector labels.Selector) ([]corev1.Node, error) {
 	if err := testclient.DataPlaneClient.List(context.TODO(), nodes, &client.ListOptions{LabelSelector: selector}); err != nil {
 		return nil, err
 	}
+
+	// If we can't find any node with worker-cnf label on hypershift, just label the first one.
+	if hypershiftutils.IsHypershiftCluster() && len(nodes.Items) == 0 {
```

Comment:
This logic cannot sit here, because here you're referring only to `worker-cnf` label while this function has general purpose to retrieve nodes by a selector - any selector no just  `worker-cnf`

Reply:
Sure I understand, it was for testing purposes only.